### PR TITLE
improve(gas-rebate) Ignore RewardsRetrieved events emitted from a batch transaction sent by designated dev account

### LINKED ISSUE #2013: Gas rebate script should ignore transactions sent from risk labs dev account
**Why**

Sometimes dev accounts will claim rewards on behalf of users (specifically when an admin proposal is passed to upgrade the DVM), the rebate program should ignore such txns.

- 3: ~1 hour

### PR DESCRIPTION
**Pull request by nicholaspai**
Signed-off-by: Nick Pai <npai.nyc@gmail.com>

**Motivation**

#2013 

Also re-runs the rebate script to encompass the latest two votes, Admin 14 and Admin 15

Therefore, this PR is a follow up the script produced by #1992 and includes a rebate for the following votes:

- Admin 10
- Admin 11
- USDETH sep expiry
- Admin 12
- Admin 13
- Admin 14
- Admin 15

**Details**
Example script run:

<details><summary>Script Details</summary>
<p>

```
➜  protocol git:(master) ✗ yarn truffle exec ./packages/core/scripts/gas-rebate/index.js --start 1598659200 --end 1600865397 --network mainnet_mnemonic
yarn run v1.22.5
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./packages/core/scripts/gas-rebate/index.js --start 1598659200 --end 1600865397 --network mainnet_mnemonic
> Warning: possible unsupported (undocumented in help) command line option: --start,--end
Using network 'mainnet_mnemonic'.



*=======================================*
*                                       *
* 🐣 Setup 🐣                           *
* - Fetching block number for timestamp *
* - Fetching historical gas px data     *
* - Fetching UMA-ETH exchange rates     *
*                                       *
*=======================================*
- Using start date: Fri Aug 28 2020 20:00:00 GMT-0400
- Using end date: Wed Sep 23 2020 08:49:57 GMT-0400
- ✅ Success, running main script now


*=======================================*
*                                       *
* 🐲⛽️ UMA Gas Rebater 🐲 ⛽️            *
*                                       *
*=======================================*
- Calculating gas rebates from block 10752294 until 10918986
- dailyAvgGasPrices: [
    {
        "timestamp": 1598572800,
        "avgGwei": "93.087752585"
    },
    {
        "timestamp": 1598659200,
        "avgGwei": "102.907957139"
    },
    {
        "timestamp": 1598745600,
        "avgGwei": "211.899164124"
    },
    {
        "timestamp": 1598832000,
        "avgGwei": "236.124357466"
    },
    {
        "timestamp": 1598918400,
        "avgGwei": "445.944863506"
    },
    {
        "timestamp": 1599004800,
        "avgGwei": "480.09630321"
    },
    {
        "timestamp": 1599091200,
        "avgGwei": "391.330863477"
    },
    {
        "timestamp": 1599177600,
        "avgGwei": "232.290794845"
    },
    {
        "timestamp": 1599264000,
        "avgGwei": "276.608264542"
    },
    {
        "timestamp": 1599350400,
        "avgGwei": "129.483046259"
    },
    {
        "timestamp": 1599436800,
        "avgGwei": "112.638419991"
    },
    {
        "timestamp": 1599523200,
        "avgGwei": "103.774996937"
    },
    {
        "timestamp": 1599609600,
        "avgGwei": "120.490254791"
    },
    {
        "timestamp": 1599696000,
        "avgGwei": "179.314983224"
    },
    {
        "timestamp": 1599782400,
        "avgGwei": "147.230410369"
    },
    {
        "timestamp": 1599868800,
        "avgGwei": "107.402741783"
    },
    {
        "timestamp": 1599955200,
        "avgGwei": "120.532907374"
    },
    {
        "timestamp": 1600041600,
        "avgGwei": "199.500184878"
    },
    {
        "timestamp": 1600128000,
        "avgGwei": "186.693081226"
    },
    {
        "timestamp": 1600214400,
        "avgGwei": "152.924068259"
    },
    {
        "timestamp": 1600300800,
        "avgGwei": "538.006886531"
    },
    {
        "timestamp": 1600387200,
        "avgGwei": "318.642907595"
    },
    {
        "timestamp": 1600473600,
        "avgGwei": "174.319934634"
    },
    {
        "timestamp": 1600560000,
        "avgGwei": "132.890442848"
    },
    {
        "timestamp": 1600646400,
        "avgGwei": "176.908105832"
    },
    {
        "timestamp": 1600732800,
        "avgGwei": "159.389045782"
    }
]
- dailyAvgUmaEthPrices: [
    {
        "timestamp": 1598572800,
        "avgPx": "0.017866339805623608"
    },
    {
        "timestamp": 1598659200,
        "avgPx": "0.022047083358610817"
    },
    {
        "timestamp": 1598745600,
        "avgPx": "0.030705415514292286"
    },
    {
        "timestamp": 1598832000,
        "avgPx": "0.028028312422093322"
    },
    {
        "timestamp": 1598918400,
        "avgPx": "0.03681999912831985"
    },
    {
        "timestamp": 1599004800,
        "avgPx": "0.051537762449157976"
    },
    {
        "timestamp": 1599091200,
        "avgPx": "0.05259979204902471"
    },
    {
        "timestamp": 1599177600,
        "avgPx": "0.053037989793070686"
    },
    {
        "timestamp": 1599264000,
        "avgPx": "0.04399597496942103"
    },
    {
        "timestamp": 1599350400,
        "avgPx": "0.04860976693515907"
    },
    {
        "timestamp": 1599436800,
        "avgPx": "0.035802267545592424"
    },
    {
        "timestamp": 1599523200,
        "avgPx": "0.04479827106142312"
    },
    {
        "timestamp": 1599609600,
        "avgPx": "0.04262813990979643"
    },
    {
        "timestamp": 1599696000,
        "avgPx": "0.04599610850913753"
    },
    {
        "timestamp": 1599782400,
        "avgPx": "0.0459634889691385"
    },
    {
        "timestamp": 1599868800,
        "avgPx": "0.04585573002364064"
    },
    {
        "timestamp": 1599955200,
        "avgPx": "0.04730263166731339"
    },
    {
        "timestamp": 1600041600,
        "avgPx": "0.04843380188933034"
    },
    {
        "timestamp": 1600128000,
        "avgPx": "0.046823441009739175"
    },
    {
        "timestamp": 1600214400,
        "avgPx": "0.04568906378892176"
    },
    {
        "timestamp": 1600300800,
        "avgPx": "0.04413669157320654"
    },
    {
        "timestamp": 1600387200,
        "avgPx": "0.03907705982124713"
    },
    {
        "timestamp": 1600473600,
        "avgPx": "0.03340700602393464"
    },
    {
        "timestamp": 1600560000,
        "avgPx": "0.030769810034934827"
    },
    {
        "timestamp": 1600646400,
        "avgPx": "0.032135681063422024"
    },
    {
        "timestamp": 1600732800,
        "avgPx": "0.035370095423556396"
    }
]


*=======================================*
*                                       *
* 🌏 Fetching Blockchain Data 🌎        *
*                                       *
*=======================================*
Reveal Events [████████████████████████████████████████] 99% | ⏳ ETA: 1s | events parsed: 240/241
Claim Events [████████████████████████████████████████] 100% | ⏳ ETA: 0s | events parsed: 233/233

*=======================================*
*                                       *
* ✅ Results                           *
*                                       *
*=======================================*
🗄 241 Reveal Transactions successfully written to /Users/nicholaspai/UMA/protocol/packages/core/scripts/gas-rebate/debug/Reveals_1.json
㊗️ Reveal Totals: {
  totalGasUsed: 54678977,
  totalEthSpent: 11.481655887864175,
  totalUmaRepaid: 274.4016780552647
}
🗄 157 Claim Transactions successfully written to /Users/nicholaspai/UMA/protocol/packages/core/scripts/gas-rebate/debug/Claims_1.json
㊗️ Claim Totals: {
  totalGasUsed: 21069297,
  totalEthSpent: 5.578421587683834,
  totalUmaRepaid: 129.03593216572423
}


*=======================================*
*                                       *
* 🧮 Final UMA Rebate                   *
*                                       *
*=======================================*
🎟 UMA to rebate: 403.4376102209888 across 66 voters
🗄  File successfully written to /Users/nicholaspai/UMA/protocol/packages/core/scripts/gas-rebate/rebates/Rebate_1.json
✨  Done in 64.96s.
```

</p>
</details>
**Issue(s)**

Fixes #2013

### CODE REVIEW

File: packages/core/scripts/gas-rebate/index.js [hunk 1]
```diff
@@ -213,39 +213,50 @@ async function parseRevealEvents({ committedVotes, revealedVotes, priceData, reb
...
-    const identifier = web3.utils.hexToUtf8(claim.returnValues.identifier);
-    const requestTime = claim.returnValues.time;
     const [transactionBlock, transactionReceipt] = await Promise.all([
       web3.eth.getBlock(claim.blockNumber),
       web3.eth.getTransactionReceipt(claim.transactionHash)
     ]);
-    const gasUsed = parseInt(transactionReceipt.gasUsed);
-    const txnTimestamp = transactionBlock.timestamp;
+    // Check if claim txn was sent by an UMA dev batch retrieval.
+    if (transactionReceipt.from !== UMA_DEV_ACCOUNT) {
```
Comment (mrice32) [inline packages/core/scripts/gas-rebate/index.js:234]: nit: maybe use toChecksumAddress on both of these addresses?

File: packages/core/scripts/gas-rebate/index.js [hunk 2]
```diff
@@ -213,39 +213,50 @@ async function parseRevealEvents({ committedVotes, revealedVotes, priceData, reb
...
+        roundId,
+        requestTime,
+        identifier,
+        claim: {
+          transactionBlock: transactionBlock.number,
+          hash: transactionReceipt.transactionHash,
+          gasUsed,
+          txnTimestamp
+        }
+      };
```
Comment (mrice32) [inline packages/core/scripts/gas-rebate/index.js:255]: How hard would it be to print out a log for each transaction that was ignored? Just so the runner can have some assurance that the script is ignoring the right things.

**Review by mrice32 — APPROVED** [review]
Looks great! Two minor comments.

File: packages/core/scripts/gas-rebate/index.js (hunk 2 above)
Comment (nicholaspai) [reply packages/core/scripts/gas-rebate/index.js:255]: I did do this but in a more roundabout way. Expand the "script details" in the Details section of the PR description.  Notice how there were 233 claim events processed, but only 157 got rebated. The diff is due to the ignores.

If its clearer I could also keep a runnning count of each claim txn ignored and print that at the end

Comment (mrice32) [reply packages/core/scripts/gas-rebate/index.js:255]: I see. Yeah, that makes sense. Didn't realize you could see that difference there. Yeah, I think that's enough for a sanity check.

**Review by mrice32 — APPROVED** [review]

**Comment by mrice32** [comment]
@nicholaspai side note: love the expandable MD you put in the description. 👌

**Pull request merged**